### PR TITLE
Add static and maximum static value

### PR DESCRIPTION
**Pull request by jseremba**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,12 +19,13 @@ https://progressbar-demo.mxapps.io/
...
 ## Usage
 Place the widget in the context of an object that has attributes for value and maximum value.
+If attribute `value` is not set, the `Value static` will be used to calculate the progress.
 
 Progress percentage is calculated as follows:
 
     (value / maximumValue) * 100
 
-If the maximum value attribute is not set, maximum value defaults to 100.
+If the maximum value attribute is not set, the maximum value defaults to `Maximum value static`.
```
Comment (McDoyen) [inline README.md]: Grammar

**Review by McDoyen — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (jseremba) [reply README.md]: Fixed

Comment (jseremba) [reply README.md]: Fixed

File: src/components/ProgressBarContainer.ts [hunk 2]
```diff
@@ -140,12 +151,37 @@ export default class ProgressBarContainer extends Component<ProgressBarContainer
...
     private updateValues(mxObject?: mendix.lib.MxObject): ProgressBarContainerState {
         return {
+            displayTextAttributeValue: mxObject ? mxObject.get(this.props.displayTextAttribute) as string : "",
             maximumValue: this.getValue(this.props.maximumValueAttribute, this.defaultMaximumValue, mxObject),
             progressValue: this.getValue(this.props.progressAttribute, undefined, mxObject),
             themeStyle: this.getBarStyle (mxObject)
         };
     }
 
+    private getDisplayText = (mxObject?: mendix.lib.MxObject) => {
```
Comment (Andries-Smit) [inline src/components/ProgressBarContainer.ts]: name of function is incorrect, "get" would expect to return something.
This function does set state... 

Would make more sense set state once, and incorporate it from `this.updateValues`

File: package.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "progressbar",
   "widgetName": "ProgressBar",
-  "version": "1.1.0",
+  "version": "1.2.0",
   "description": "Displays a progress bar",
```
Comment (Andries-Smit) [inline package.json:5]: Please update all dependencies too

File: src/ProgressBar.xml [hunk 4]
```diff
@@ -6,29 +6,39 @@
...
     </icon>
     <properties>
```
Comment (Andries-Smit) [inline src/ProgressBar.xml:8]: MxTestProject is not up to date.

File: src/ProgressBar.xml [hunk 5]
```diff
@@ -6,29 +6,39 @@
...
         </property>
+        <property key="staticMaximumValue" type="integer" defaultValue="100">
+            <caption>Maximum value static</caption>
+            <category>Values</category>
+            <description/>
+        </property>
         <property key="bootstrapStyleAttribute" type="attribute" required="false">
             <caption>Bar style attribute</caption>
-            <category>Data source</category>
+            <category>Values</category>
```
Comment (Andries-Smit) [inline src/ProgressBar.xml]: Move to category 'Appearance', after 'Bar style'

File: src/ProgressBar.webmodeler.ts [hunk 6]
```diff
@@ -29,11 +29,22 @@ export class preview extends Component<ProgressBarContainerProps, {}> {
...
+
+    private getDisplayTextValue() {
+        if (this.props.displayText === "attribute") {
+            return `{ ${this.props.displayTextAttribute} }`;
+        } else if (this.props.displayText === "static") {
+            return this.props.displayTextStatic;
+        }
+
+        return "";
+    }
```
Comment (Andries-Smit) [inline src/ProgressBar.webmodeler.ts:48]: Missing the `getVisibleProperties` changes like https://github.com/FlockOfBirds/progress-circle/pull/12/files#diff-e8e1d3f637e5239aba496b065ce6c355R53

File: src/components/ProgressBar.ts [hunk 7]
```diff
@@ -75,12 +79,24 @@ class ProgressBar extends Component<ProgressBarProps, {}> {
         return Math.round((progress / maximumValue) * 100);
     }
 
-    private getProgressText(maximumValue: number, progress?: number): string {
-        if (progress || progress === 0) {
-            return maximumValue < 1 ? "Invalid" : `${this.calculatePercentage(progress, maximumValue)}%`;
+    private setProgress(maximumValue: number, value: number | undefined, text?: DisplayText, displayTextValue?: string): string {
```
Comment (Andries-Smit) [inline src/components/ProgressBar.ts]: function name does not match wat it is doing... `set<Some>` implies that nothing is returned...

File: src/components/ProgressBarContainer.ts [hunk 8]
```diff
@@ -88,18 +96,21 @@ export default class ProgressBarContainer extends Component<ProgressBarContainer
...
-            progress: this.state.progressValue,
+            progress: this.props.progressAttribute ? this.state.progressValue || 0 : this.props.staticValue,
             style: ProgressBarContainer.parseStyle(this.props.style)
         });
     }
 
     componentWillReceiveProps(newProps: ProgressBarContainerProps) {
         this.resetSubscription(newProps.mxObject);
         this.setState(this.updateValues(newProps.mxObject));
+        this.getDisplayText(newProps.mxObject);
```
Comment (Andries-Smit) [inline src/components/ProgressBarContainer.ts]: should be joint with `updateValues`

File: src/components/ProgressBarContainer.ts [hunk 9]
```diff
@@ -140,12 +151,37 @@ export default class ProgressBarContainer extends Component<ProgressBarContainer
...
+        const { displayTextAttribute } = this.props;
+
+        if (mxObject) {
+            this.setState({
+                displayTextAttributeValue: mxObject.get(displayTextAttribute) as string
+            });
+        } else {
+            this.setState({
+                displayTextAttributeValue: ""
+            });
```
Comment (Andries-Smit) [inline src/components/ProgressBarContainer.ts]: useless if else statement could be done with ternary operator
`mxObject ? mxObject.get(displayTextAttribute) as string : ""`

**Review by Andries-Smit — CHANGES_REQUESTED** [review]

File: src/components/ProgressBarContainer.ts (hunk 2 above)
Comment (jseremba) [reply src/components/ProgressBarContainer.ts]: Fixed

File: src/ProgressBar.xml (hunk 5 above)
Comment (jseremba) [reply src/ProgressBar.xml]: Done

File: src/components/ProgressBarContainer.ts (hunk 8 above)
Comment (jseremba) [reply src/components/ProgressBarContainer.ts]: Done

File: src/components/ProgressBarContainer.ts (hunk 9 above)
Comment (jseremba) [reply src/components/ProgressBarContainer.ts]: Done

File: src/ProgressBar.webmodeler.ts (hunk 6 above)
Comment (jseremba) [reply src/ProgressBar.webmodeler.ts:48]: Added

File: src/components/ProgressBar.ts (hunk 7 above)
Comment (jseremba) [reply src/components/ProgressBar.ts]: Renamed

File: src/ProgressBar.xml (hunk 4 above)
Comment (jseremba) [reply src/ProgressBar.xml:8]: Updated

File: src/components/ProgressBarContainer.ts (hunk 2 above)
Comment (jseremba) [reply src/components/ProgressBarContainer.ts]: Changed

File: package.json (hunk 3 above)
Comment (jseremba) [reply package.json:5]: Done

**Pull request merged**